libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/SslConfigurationLoader.java

```diff
@@ -281,21 +287,22 @@ public SslConfiguration load(Path basePath) {
         final SslVerificationMode verificationMode = resolveSetting(VERIFICATION_MODE, SslVerificationMode::parse, defaultVerificationMode);
         final SslClientAuthenticationMode clientAuth = resolveSetting(CLIENT_AUTH, SslClientAuthenticationMode::parse, defaultClientAuth);
 
-        final SslTrustConfig trustConfig = buildTrustConfig(basePath, verificationMode);
         final SslKeyConfig keyConfig = buildKeyConfig(basePath);
+        final SslTrustConfig trustConfig = buildTrustConfig(basePath, verificationMode, keyConfig);
 
         if (protocols == null || protocols.isEmpty()) {
             throw new SslConfigException("no protocols configured in [" + settingPrefix + PROTOCOLS + "]");
         }
         if (ciphers == null || ciphers.isEmpty()) {
             throw new SslConfigException("no cipher suites configured in [" + settingPrefix + CIPHERS + "]");
         }
-        return new SslConfiguration(trustConfig, keyConfig, verificationMode, clientAuth, ciphers, protocols);
+        final boolean isExplicitlyConfigured = hasSettings(settingPrefix);
+        return new SslConfiguration(isExplicitlyConfigured, trustConfig, keyConfig, verificationMode, clientAuth, ciphers, protocols);
```

Comment:
`isExplicitlyConfigured` is used in watcher to know whether an SSL configuration really exists. It always "exists" - the SSL service always creates a context, but watcher wants to know whether the admin actually configured it, or whether the SSL service created a default one.